tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -238,3 +271,174 @@ def get_retention_from_backup_policy(self, retention_policy):
+        op=None
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(ShortTermBackupRetentionPolicyAction, self).__init__(
+            BackupRetentionPolicyHelper.SHORT_TERM_SQL_OPERATIONS, *args, **kwargs)
+        self.retention_period_days = self.data['retention-period-days']
+
+    def validate(self):
+        if self.retention_period_days not in \
+                ShortTermBackupRetentionPolicyAction.VALID_RETENTION_PERIOD_DAYS:
+            raise PolicyValidationError(
+                "Invalid retention-period-days: {}. Valid values are: {}".format(
+                    self.retention_period_days,
+                    ShortTermBackupRetentionPolicyAction.VALID_RETENTION_PERIOD_DAYS
+                )
+            )
+        return self
+
+    def get_parameters_for_new_retention_policy(self, database):
+        return self.retention_period_days
+
+
+@SqlDatabase.action_registry.register('update-long-term-backup-retention-policy')
+class LongTermBackupRetentionPolicyAction(BackupRetentionPolicyAction):
+    """
+
+    Update the long term backup retention policy for a SQL Database.
+
+    There are 3 backup types for a sql database: weekly, monthly, and yearly. And, each
+    of these backups has a retention period that can specified in units of days, weeks,
+    months, or years.
+
+    :example: Enforce a 1 month maximum retention for weekly backups on all SQL Databases
+
+    .. code-block:: yaml
+
+            policies:
+              - name: update-long-term-backup-retention-policy
+                resource: azure.sqldatabase
+                filters:
+                  - type: long-term-backup-retention-policy
+                    backup-type: weekly
+                    op: gt
+                    retention-period: 1
+                    retention-period-units: months
+                actions:
+                  - type: update-long-term-backup-retention-policy
+                    backup-type: weekly
+                    retention-period: 1
+                    retention-period-units: months
+
+    """
+
+    schema = type_schema(
+        'update-long-term-backup-retention-policy',
+        rinherit=LongTermBackupRetentionPolicyFilter.schema,
+        op=None
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(LongTermBackupRetentionPolicyAction, self).__init__(
+            BackupRetentionPolicyHelper.LONG_TERM_SQL_OPERATIONS, *args, **kwargs)
+        self.backup_type = BackupRetentionPolicyHelper.LongTermBackupType[self.data.get(
+            'backup-type')]
+        retention_period = self.data['retention-period']
+        retention_period_units = RetentionPeriod.Units[self.data['retention-period-units']]
+        self.iso8601_duration = RetentionPeriod.iso8601_duration_from_period_and_units(
+            retention_period,
+            retention_period_units
+        )
+
+    def get_parameters_for_new_retention_policy(self, database):
+        client = self.manager.get_client()
```

Comment:
We might have cached retention policy, right?

Reply:
yeah, the caching is handled internally in the `BackupRetentionPolicyHelper.get_backup_retention_policy()` call.

Reply:
oh, ok. Can we create a client in `__init__` to avoid doing it for each database?